app/services/DynamoChannelTests.scala

```diff
@@ -47,32 +49,56 @@ class DynamoChannelTests(stage: String) extends StrictLogging {
       DynamoGetError(error)
     )
 
-  private def putAll(items: List[java.util.Map[String, AttributeValue]]): ZIO[Blocking, DynamoPutError, BatchWriteItemResponse] =
+  // Sends a batch of write requests, and returns any unprocessed items
+  private def putAll(writeRequests: List[WriteRequest]): ZIO[ZEnv, DynamoPutError, List[WriteRequest]] =
     effectBlocking {
-      val writeRequests = items.map(item =>
-        WriteRequest.builder.putRequest(
-          PutRequest
-            .builder
-            .item(item)
-            .build()
-        ).build()
-      )
       val batchWriteRequest =
         BatchWriteItemRequest
           .builder
           .requestItems(Map(tableName -> writeRequests.asJava).asJava)
           .build()
 
       val result = client.batchWriteItem(batchWriteRequest)
-      logger.info(s"Batch write result: $result")
-      // TODO - handle any unprocessedItems
-      result
+      logger.info(s"BatchWriteItemResponse: $result")
+
+      result.unprocessedItems().asScala
+        .get(tableName)
+        .map(items => items.asScala.toList)
+        .getOrElse(Nil)
 
     }.mapError(error =>
       DynamoPutError(error)
     )
 
-  def getAllTests[T <: ChannelTest[T] : Decoder](channel: Channel): ZIO[Blocking, DynamoGetError, List[T]] =
+  /**
```

Comment:
@adamnfish any thoughts on `putAllBatched`? The use of an empty stream feels a bit odd, but it gives me all the features I wanted